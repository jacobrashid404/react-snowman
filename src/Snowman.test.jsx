import { test, expect } from "vitest";
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman.jsx";

test("If loss, can't click on board, and image doesn't crash", function () {
  const { container } = render(<Snowman />); // TODO: add this to test -> words = ["apple"], maxWrong= 6
  const lettersToCheck = ["w", "x", "y", "z", "b", "c", "k"];

  const $containerElems = lettersToCheck.map((ltr) => { //FIXME: fix this test
    const elem = container.querySelector(`.letter_${ltr}`);
    const elem = container.querySelector(`button[value="${ltr}"]`); //CSS VALUE SELECTOR!!!!

    try {
      fireEvent.click(elem);
    } catch (err) {
      expect(err.message).toContain("Unable to fire a");
    }
    return elem;
  });

  const snowmanImg = container.querySelector(".Snowman-img");
  expect(snowmanImg.src).toContain("png");
});

//TODO: same test but lose at a different value

//TODO: real verification of functionality, do we win?

//TODO: some more testing!

test("snapshot test", function () {
  const { container } = render(<Snowman />);
  expect(container).toMatchSnapshot();
});
