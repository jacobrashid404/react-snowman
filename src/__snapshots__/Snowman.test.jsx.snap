// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="0"
      class="Snowman-img"
      src="/src/0.png"
    />
    <p>
      Wrong guesses: 
      0
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p>
      <button
        class="letter_a"
        value="a"
      >
        a
      </button>
      <button
        class="letter_b"
        value="b"
      >
        b
      </button>
      <button
        class="letter_c"
        value="c"
      >
        c
      </button>
      <button
        class="letter_d"
        value="d"
      >
        d
      </button>
      <button
        class="letter_e"
        value="e"
      >
        e
      </button>
      <button
        class="letter_f"
        value="f"
      >
        f
      </button>
      <button
        class="letter_g"
        value="g"
      >
        g
      </button>
      <button
        class="letter_h"
        value="h"
      >
        h
      </button>
      <button
        class="letter_i"
        value="i"
      >
        i
      </button>
      <button
        class="letter_j"
        value="j"
      >
        j
      </button>
      <button
        class="letter_k"
        value="k"
      >
        k
      </button>
      <button
        class="letter_l"
        value="l"
      >
        l
      </button>
      <button
        class="letter_m"
        value="m"
      >
        m
      </button>
      <button
        class="letter_n"
        value="n"
      >
        n
      </button>
      <button
        class="letter_o"
        value="o"
      >
        o
      </button>
      <button
        class="letter_p"
        value="p"
      >
        p
      </button>
      <button
        class="letter_q"
        value="q"
      >
        q
      </button>
      <button
        class="letter_r"
        value="r"
      >
        r
      </button>
      <button
        class="letter_s"
        value="s"
      >
        s
      </button>
      <button
        class="letter_t"
        value="t"
      >
        t
      </button>
      <button
        class="letter_u"
        value="u"
      >
        u
      </button>
      <button
        class="letter_v"
        value="v"
      >
        v
      </button>
      <button
        class="letter_w"
        value="w"
      >
        w
      </button>
      <button
        class="letter_x"
        value="x"
      >
        x
      </button>
      <button
        class="letter_y"
        value="y"
      >
        y
      </button>
      <button
        class="letter_z"
        value="z"
      >
        z
      </button>
    </p>
    <button
      class="btn btn-primary"
    >
      Restart Game
    </button>
  </div>
</div>
`;
